core/src/cost_tracking_service.rs

```diff
@@ -0,0 +1,185 @@
+//! this service receives transactions that have been successfully added
+//! to a bank from banking_stage, it therefore calculates and applies the
+//! costs of these transactions to cost_tracker.
+//! This process has some overhead, being in its own service thread is to
+//! minimize impact on main TPU threads.
+
+use crate::cost_tracker::CostTracker;
+use solana_measure::measure::Measure;
+use solana_runtime::bank::TransactionExecutionResult;
+use solana_sdk::{timing::timestamp, transaction::Transaction};
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        mpsc::Receiver,
+        Arc, RwLock,
+    },
+    thread::{self, Builder, JoinHandle},
+    time::Duration,
+};
+
+#[derive(Default)]
+pub struct CostTrackingServiceStats {
+    last_print: u64,
+    cost_tracker_update_count: u64,
+    cost_tracker_update_elapsed: u64,
+}
+
+impl CostTrackingServiceStats {
+    fn update(&mut self, cost_tracker_update_count: u64, cost_tracker_update_elapsed: u64) {
+        self.cost_tracker_update_count += cost_tracker_update_count;
+        self.cost_tracker_update_elapsed += cost_tracker_update_elapsed;
+
+        let now = timestamp();
+        let elapsed_ms = now - self.last_print;
+        if elapsed_ms > 1000 {
+            datapoint_info!(
+                "cost-tracking-service-stats",
+                ("total_elapsed_us", elapsed_ms * 1000, i64),
```

Comment:
Is this one necessary? Seems like this is measuring the time between datapoints which

1) Should update roughly every two seconds  based on the check `if elapsed_ms > 1000` ,
2)  The datapoints themselves are already timestamped on arrival to Influx

Reply:
Oops, a copy/paste from `banking_stage.rs`, will remove